project/src/test/java/com/google/sps/EventSpannerTasksTest.java

```diff
@@ -0,0 +1,121 @@
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    // Mock a request to trigger the SpannerClient setup to run
+    MockServletContext mockServletContext = new MockServletContext();
+    new SpannerClient().contextInitialized(new ServletContextEvent(mockServletContext));
+    SpannerTestTasks.setup();
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    SpannerTestTasks.cleanup();
+  }
+
+
+  /** Verify insertion of event in db and retrieval by id 
+   * Also tests behavior of EventCreationServlet doGet() where doGet request.getParameter("eventId") == event.getId() */
+  @Test
+  public void eventInsertAndRetrieval() {
+    SpannerTasks.insertOrUpdateUser(HOST);
+    Event event =
+        new Event.Builder(EVENT_NAME, DESCRIPTION, LABELS, LOCATION, DATE, TIME, HOST).build();
+    SpannerTasks.insertorUpdateEvent(event);
+    Event dbEvent = SpannerTasks.getEventById(event.getId()).get();
+
+    Assert.assertEquals(dbEvent.getName(), EVENT_NAME);
+    Assert.assertEquals(dbEvent.getDescription(), DESCRIPTION);
+    Assert.assertEquals(dbEvent.getLabels(), LABELS);
+    Assert.assertEquals(dbEvent.getLocation(), LOCATION);
+    Assert.assertEquals(dbEvent.getDate(), DATE);
+    Assert.assertEquals(dbEvent.getTime(), TIME);
+    Assert.assertEquals(dbEvent.getHost(), HOST);
+  }
+
+  /** Verify getting Set<Event> from corresponding eventIds */
+  @Test
+  public void getEventsByIdTest() {
+    SpannerTasks.insertOrUpdateUser(HOST);
+    Event event =
+        new Event.Builder(EVENT_NAME, DESCRIPTION, LABELS, LOCATION, DATE, TIME, HOST).build();
+    SpannerTasks.insertorUpdateEvent(event);
+    Event otherEvent =
+        new Event.Builder(NEW_EVENT_NAME, DESCRIPTION, LABELS, LOCATION, DATE, TIME, HOST).build();
+    SpannerTasks.insertorUpdateEvent(otherEvent);
+    Set<Event> insertedEvents = new HashSet<>(Arrays.asList(event, otherEvent));
+
+    Set<Event> dbEvents = SpannerTasks.getEventsFromIds(Arrays.asList(event.getId(), otherEvent.getId()));
+
+    Assert.assertEquals(dbEvents, insertedEvents);
+  }
+
+  /**
+   * Verify putting event in database through doPost with correct params and getting back correct
+   * redirectURL
+   */
+  @Test
+  public void testEventCreationDoPost() throws Exception {
+    MockHttpServletRequest request = new MockHttpServletRequest();
+    MockHttpServletResponse response = new MockHttpServletResponse();
+    request.addParameter("name", EVENT_NAME);
+    request.addParameter("date", DATE_STRING);
+    request.addParameter("time", TIME);
+    request.addParameter("description", DESCRIPTION);
+    request.addParameter("location", LOCATION);
+
+    new EventCreationServlet().doPost(request, response);
+
+    // Get back Event posted in db
+    Event returnEvent = new Gson().fromJson(response.getContentAsString(), Event.class);
+
+    // Check redirected URL is the ID of the item put in as request
+    Assert.assertEquals(
```

Comment:
is it ok to assert the redirectUrl? (which is what is ultimately done in the doPost()) or should i create a new event based on what is given back and compare those using  Assert.assertEquals?

Reply:
I think you should also test if the event is in DB.